pkg/storage/stores/shipper/indexshipper/tsdb/identifier.go

```diff
@@ -77,9 +81,13 @@ func (i SingleTenantTSDBIdentifier) Hash(h hash.Hash32) (err error) {
 
 // str builds filename with format <file-creation-ts> + `-` + `compactor` + `-` + <oldest-chunk-start-ts> + `-` + <latest-chunk-end-ts> `-` + <index-checksum>
 func (i SingleTenantTSDBIdentifier) str() string {
+	ts := i.origTS
+	if ts == 0 {
+		ts = i.TS.UnixNano()
```

Comment:
nit: I find this a bit hacky. I think a bool flag `writeTSAsSecs`  would read better.